src/devices/QwiicButton/QwiicButton.cs

```diff
@@ -0,0 +1,96 @@
+﻿//// Licensed to the .NET Foundation under one or more agreements.
+//// The .NET Foundation licenses this file to you under the MIT license.
+//// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+using Iot.Device.Common;
+using Iot.Device.QwiicButton.RegisterMapping;
+
+namespace Iot.Device.QwiicButton
+{
+    /// <summary>
+    /// SparkFun Qwiic Button is an I2C based button with a built-in LED.
+    /// Supported hardware version: 1.0.0
+    /// </summary>
+    public sealed partial class QwiicButton : IDisposable
+    {
+        private const int DefaultAddress = 0x6F; // Default I2C address of the button
+        private I2cRegisterAccess<Register> _registerAccess;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QwiicButton"/> class.
+        /// </summary>
+        /// <param name="i2cBusId">I2C bus ID the button is connected to.</param>
+        /// <param name="i2cAddress">I2C bus address of the button (default=0x6F).</param>
+        public QwiicButton(int i2cBusId, byte i2cAddress = DefaultAddress)
+        {
+            I2cBusId = i2cBusId;
```

Comment:
why do you need to store those two? You should at minimum provide constructor which takes I2cDevice - some I2CDevices are provided using i.e. Ft4222 and they can't be constructed using I2cDevice.Create

Reply:
Removed